skyportal/models.py

```diff
@@ -725,6 +742,24 @@ class Comment(Base):
         passive_deletes=True,
     )
 
+    @classmethod
+    def get_if_owned_by(cls, ident, user, options=[]):
+        comment = cls.query.options(options).get(ident)
+
+        if comment is not None and not comment.is_owned_by(user):
+            raise AccessError('Insufficient permissions.')
+
+        # Grab basic author info for the comment
+        author = User.query.filter(User.username == comment.author).first()
+        comment.author_info = {
```

Comment:
Why not then pass `author` to `author_info`, instead of duplicating the dictionary?

Otherwise, perhaps:

```
user_info = {field: getattr(user, field) for field in ('username', 'first_name', 'last_name', 'gravatar_url')}
```

Or consider `user.to_dict()`?

Reply:
I mainly didn't want to pass along the entire User model as that would technically expose contact_emails for every author with comments on a given source to anyone with access to that source and I wasn't sure if that was something we wanted. Also, it never hurts to lower the size of messages we're passing between the back-end and front-end. 

That said, I like the use of getattr() to make that part more succinct so I've incorporated that in.